### PR TITLE
show flutter plugins as IntelliJ libraries

### PR DESCRIPTION
**Pull request by devoncarew**
- show flutter plugins as IntelliJ libraries (fix https://github.com/flutter/flutter-intellij/issues/1090)

This shows shows flutter plugins in the `External libraries` section of the Project view ,as well as in the project structure dialog. You can expand these entries to dig into the android and ios code that makes up the plugin implementation. Some screenshots:

screen shot 2017-09-21 at 1 32 04 pm

screen shot 2017-09-21 at 1 32 15 pm

screen shot 2017-09-21 at 1 32 53 pm

@stevemessick @pq

### CODE REVIEW

File: src/io/flutter/pub/PubRoot.java [hunk 1]
```diff
@@ -264,9 +275,52 @@ public VirtualFile getPubspec() {
    * Returns true if the pubspec declares a flutter dependency.
    */
   public boolean declaresFlutter() {
+    // It uses FLutter if it contains:
```
Comment (pq) [inline src/io/flutter/pub/PubRoot.java]: FLutter => Flutter

File: src/io/flutter/sdk/FlutterPluginsLibraryManager.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+    final VirtualFile parent = file.getParent();
+    final VirtualFile pubspec = parent == null ? null : parent.findChild(PUBSPEC_YAML);
+
+    if (pubspec != null) {
+      scheduleUpdate();
+    }
+  }
+
+  private final AtomicBoolean isUpdating = new AtomicBoolean(false);
```
Comment (stevemessick) [inline src/io/flutter/sdk/FlutterPluginsLibraryManager.java]: Do we have a convention about placement of declarations? I always put fields at the top of a class, and tend to expect to see them there. Actually, I usually reformat + reorganize code, which puts things in order.

**Comment by devoncarew** [comment]
> Very cool! Are you planning to add packages, too?

Thanks! The Dart IntelliJ plugin already shows packages; this specialization just shows flutter plugins (and shows the full package source, not just the lib folder).

File: src/io/flutter/sdk/FlutterPluginsLibraryManager.java (hunk 2 above)
Comment (devoncarew) [reply src/io/flutter/sdk/FlutterPluginsLibraryManager.java]: I'll sort this nearer to the top -

**Pull request merged**